### PR TITLE
juno: Fix the number of rates returns for triplet rate array format

### PR DESCRIPTION
**Pull request by sudeep-holla**
As per the specification, the number of rates should be 3 if the triplet rate
array format(bit[12]) is set.

Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>
Change-Id: Ia7c3299b8327ec0a50923941c1e9a11e80fcc504

### CODE REVIEW

File: module/scmi_clock/src/mod_scmi_clock.c [hunk 1]
```diff
@@ -1291,7 +1291,7 @@ static int scmi_clock_describe_rates_handler(fwk_id_t service_id,
         return_values.num_rates_flags =
             SCMI_CLOCK_DESCRIBE_RATES_NUM_RATES_FLAGS(
                 /* Only a single rate is returned */
-                1U,
+                3U,
```
[the remark this answers is not in the document]
Comment (sudeep-holla) [reply module/scmi_clock/src/mod_scmi_clock.c:1294]: I am fine with that, I can see how it is confusing. Do you want me to refresh the patch or you with all maintainer powers can manage to do that yourself ? Let me know if you still prefer me to do that.

**Comment by sudeep-holla** [comment]
```
     CLOCK ID: 3
     [Check 1] Query describe rates
       CHECK STATUS   : PASSED [SCMI_STATUS_SUCCES] 
       CHECK HEADER   : PASSED [0x00005004]        
       CHECK RSVD BITS: PASSED                     
       CHECK NUM OF RATES        : FAILED
         EXPECTED   : 0x00000001                
         RECEIVED   : 0x00000003                          : NON CONFORMANT
```
The conformance test suite needs to be updated. Now we know the reason why this was not caught earlier.

**Comment by tarek-arm** [comment]
Jenkins: OK to test

**Comment by brett-warren-arm** [comment]
retest this please

**Review by tarek-arm — APPROVED** [review]

**Pull request merged**